apps/server/src/services/rundown-service/__tests__/delayUtils.test.ts

```diff
@@ -1,249 +1,221 @@
+      { id: '3', type: SupportedEvent.Block } as OntimeBlock,
+      makeOntimeEvent({ id: '4', timeStart: 20, timeEnd: 30, duration: 10, linkStart: null }),
+      makeOntimeEvent({ id: '5', timeStart: 30, timeEnd: 40, duration: 10, linkStart: '4' }),
+    ];
+
+    const updatedRundown = apply('delay', testRundown);
+    expect(updatedRundown).not.toBe(testRundown);
+    expect(updatedRundown).toMatchObject([
+      { id: '1', timeStart: 10, timeEnd: 20, duration: 10, revision: 2 },
+      { id: '2', timeStart: 20, timeEnd: 30, duration: 10, revision: 2, linkStart: '1' },
+      { id: '3' },
+      { id: '4', timeStart: 30, timeEnd: 40, duration: 10, revision: 2, linkStart: null },
+      { id: '5', timeStart: 40, timeEnd: 50, duration: 10, revision: 2, linkStart: '4' },
+    ]);
+  });
+
+  it('applies negative delays', () => {
+    const testRundown = [
+      makeOntimeDelay(-10),
+      makeOntimeEvent({ id: '1', timeStart: 0, timeEnd: 10, duration: 10 }),
+      makeOntimeEvent({ id: '2', timeStart: 10, timeEnd: 20, duration: 10, linkStart: '1' }),
+      { id: '3', type: SupportedEvent.Block } as OntimeBlock,
+      makeOntimeEvent({ id: '4', timeStart: 20, timeEnd: 30, duration: 10, linkStart: null }),
+      makeOntimeEvent({ id: '5', timeStart: 30, timeEnd: 40, duration: 10, linkStart: '4' }),
+    ];
+
+    const updatedRundown = apply('delay', testRundown);
+    expect(updatedRundown).toMatchObject([
+      { id: '1', timeStart: 0, timeEnd: 10, duration: 10, revision: 2 },
+      { id: '2', timeStart: 0, timeEnd: 10, duration: 10, revision: 2, linkStart: null },
+      { id: '3' },
+      { id: '4', timeStart: 10, timeEnd: 20, duration: 10, revision: 2, linkStart: null },
+      { id: '5', timeStart: 20, timeEnd: 30, duration: 10, revision: 2, linkStart: '4' },
+    ]);
+  });
+
+  it('should account for minimum duration and start when applying negative delays', () => {
+    const testRundown: OntimeRundown = [
+      makeOntimeDelay(-50),
+      makeOntimeEvent({ id: '1', timeStart: 0, timeEnd: 100, duration: 100 }),
+      makeOntimeEvent({ id: '2', timeStart: 100, timeEnd: 150, duration: 50, linkStart: '1' }),
+    ];
+
+    const expected = [
+      { id: '1', type: SupportedEvent.Event, timeStart: 0, timeEnd: 100, duration: 100, revision: 2 } as OntimeEvent,
+      {
+        id: '2',
+        type: SupportedEvent.Event,
+        timeStart: 50,
+        timeEnd: 100,
+        duration: 50,
+        linkStart: null,
+        revision: 2,
+      } as OntimeEvent,
+    ];
+
+    const updatedRundown = apply('delay', testRundown);
+    expect(updatedRundown).toMatchObject(expected);
+  });
+
+  it('unlinks events to maintain gaps when applying positive delays', () => {
+    const testRundown = [
+      makeOntimeEvent({ id: '1', timeStart: 0, timeEnd: 100, duration: 100, revision: 1 }),
+      makeOntimeDelay(50),
+      makeOntimeEvent({ id: '2', timeStart: 100, timeEnd: 150, duration: 50, revision: 1, linkStart: '1' }),
+    ];
+
```

Comment:
test for falling case
```suggestion
  it('unlinks events to across blocks is it is the first event after the delay', () => {
    const testRundown = [
      makeOntimeEvent({ id: '1', timeStart: 0, timeEnd: 100, duration: 100, revision: 1 }),
      makeOntimeDelay(50),
      makeOntimeBlock('BLOCK'),
      makeOntimeEvent({ id: '2', timeStart: 100, timeEnd: 150, duration: 50, revision: 1, linkStart: '1' }),
    ];

    expect(apply('delay', testRundown)).toMatchObject([
      { id: '1', type: SupportedEvent.Event, timeStart: 0, timeEnd: 100, duration: 100, revision: 1 } as OntimeEvent,
      { id: 'block', type: SupportedEvent.Block, title: 'BLOCK' },
      {
        id: '2',
        type: SupportedEvent.Event,
        timeStart: 150,
        timeEnd: 200,
        duration: 50,
        linkStart: null,
        revision: 2,
      } as OntimeEvent,
    ]);
  });
```